src/OperatorMsg.chpl

```diff
@@ -194,10 +194,10 @@ module OperatorMsg
             var l = toSymEntry(left,uint);
             var r = toSymEntry(right,bool);
             if boolOps.contains(op) {
-              var e = st.addEntry(rname, l.size, bool);
+              var e = st.addEntry(rname, l.size, uint);
               return doBinOpvv(l, r, e, op, rname, pn, st);
             }
-            var e = st.addEntry(rname, l.size, uint);
+            var e = st.addEntry(rname, l.size, real);
```

Comment:
We should be returning a `uint` here not a `real`.

Reply:
Yup, I changed the wrong one. Let me fix it rq